tests/ui/json/json-short.stderr

```diff
@@ -13,7 +13,7 @@ If you don't know the basics of Rust, you can look at the
 [Rust Book][rust-book] to get started.
 
 [rust-book]: https://doc.rust-lang.org/book/
-"},"level":"error","spans":[{"file_name":"$DIR/json-short.rs","byte_start":63,"byte_end":63,"line_start":1,"line_end":1,"column_start":64,"column_end":64,"is_primary":true,"text":[{"text":"//@ compile-flags: --json=diagnostic-short --error-format=json","highlight_start":64,"highlight_end":64}],"label":"consider adding a `main` function to `$DIR/json-short.rs`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"$DIR/json-short.rs:1:64: error[E0601]: `main` function not found in crate `json_short`: consider adding a `main` function to `$DIR/json-short.rs`
+"},"level":"error","spans":[{"file_name":"$DIR/json-short.rs","byte_start":121,"byte_end":121,"line_start":2,"line_end":2,"column_start":59,"column_end":59,"is_primary":true,"text":[{"text":"
```

Comment:
Discussion: same here, it seems like we lost the suggestion parts 🤔

Reply:
I think normalization that removes `//~` annotations from `.stderr` files messes up the json output here.
All the cases here and in other tests can be fixed by moving the annotations to different lines.